### PR TITLE
Enable RedundantNamespace Lint

### LINKED ISSUE #10480: Enable Lint: `RedundantNamespace`
We perform lint checking to ensure code correctness and improve performance, but many of the system lint checks have not been enabled.

The goal of this issue is to enable the lint check: `RedundantNamespace`

Line: https://github.com/ankidroid/Anki-Android/blob/6260cd6e25f67f331833bbd0981f2cda5a426dde/lint-release.xml#L74

To do this: 

* Change `ignore` to `fatal`
* Move the line in the XML to be somewhere near the top of the file (before the large block of `ignore` elements. 
  * The exact position is up to you, but if there's no comment associated with the check, a large nearly alphabetical block of `fatal` warnings seems sensible
* Run `gradlew lint` in the Android Studio terminal and fix the errors which occur.
  * `@Suppress(str)` (Kotlin) or `@SuppressWarnings(str)`  (Java) the warnings if they are valid. Android Studio's quickfixes menu will offer options to suppress
    * Use your judgment, but aim to suppress as close to the error as possible (line/method) unless the class specifically has a reason why it will violate a lint check many times
  * If it was an error that we missed, fix it.
* Submit a Pull Request

### PR DESCRIPTION
**Pull request by Prince-kushwaha**
## Purpose / Description

Enable RedundantNamespace Lint

## Fixes
Fixes #10480 
## Approach
 issue id="RedundantNamespace" severity="ignore" 
Change ignore to fatal
fix the issues

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: .idea/codeStyles/Project.xml [hunk 1]
```diff
@@ -51,26 +51,8 @@
       </option>
     </JavaCodeStyleSettings>
     <JetCodeStyleSettings>
-      <option name="PACKAGES_TO_USE_STAR_IMPORTS">
```
Comment (david-allison) [inline .idea/codeStyles/Project.xml:54]: This shouldn't be changed

File: .idea/codeStyles/Project.xml [hunk 2]
```diff
@@ -238,5 +220,8 @@
         </rules>
       </arrangement>
     </codeStyleSettings>
+    <codeStyleSettings language="kotlin">
+      <option name="CODE_STYLE_DEFAULTS" value="KOTLIN_OFFICIAL" />
```
Comment (david-allison) [inline .idea/codeStyles/Project.xml:224]: same

File: lint-release.xml [hunk 3]
```diff
@@ -1,5 +1,4 @@
-<?xml version="1.0" encoding="UTF-8"?>
-    <!-- INFORMATION ABOUT USAGE
+<?xml version="1.0" encoding="UTF-8"?><!-- INFORMATION ABOUT USAGE
```
Comment (david-allison) [inline lint-release.xml:1]: did this need to change? it doesn't seem relevant

File: lint-release.xml [hunk 4]
```diff
@@ -51,6 +50,9 @@
     <issue id="RtlEnabled" severity="fatal" />
     <issue id="RtlHardcoded" severity="fatal" />
 
+    <issue id="RedundantNamespace" severity="fatal" />
```
Comment (david-allison) [inline lint-release.xml:53]: Awesome!

**Review by david-allison — CHANGES_REQUESTED** [review]
Thanks!! Looks good, a few questions

File: .idea/codeStyles/Project.xml (hunk 1 above)
Comment (Prince-kushwaha) [reply .idea/codeStyles/Project.xml]: sorry it done by mistake

File: .idea/codeStyles/Project.xml (hunk 2 above)
Comment (Prince-kushwaha) [reply .idea/codeStyles/Project.xml]: it done by mistake i undo the changes

File: lint-release.xml (hunk 3 above)
Comment (Prince-kushwaha) [reply lint-release.xml:1]: it done by mistake i undo the changes

**Review by david-allison — CHANGES_REQUESTED** [review]
One minor change: `<?xml version="1.0" encoding="UTF-8"?><!-- INFORMATION ABOUT USAGE`

Rest is fantastic

**Review by david-allison — APPROVED** [review]
Looks great, thanks!

**Review by Akshay0701 — APPROVED** [review]
LGTM

**Comment by mikehardy** [comment]
took a conflict from a related change - lots of lint additions right now so that is a hot spot on the file, probably just need to move it into place again, but without losing the other lint rule change to fatal of course

**Comment by Prince-kushwaha** [comment]
@ShridharGoel i do not understand what you want to say?

**Comment by ShridharGoel** [comment]
https://www.git-tower.com/learn/git/faq/git-squash
https://www.baeldung.com/ops/git-squash-commits

**Review by mikehardy — APPROVED** [review]
The changes, when looked at as a single unit, are good - thank you!
We can squash merge these, I'm okay with that
However, when you say "no" to a request that is related to source code management, you must realize that the actual use of the source code tools (git, and learning how to do rebases and squashes in git) is a very very powerful area of knowledge, and it's really important when interacting with large open source projects since reviewer time is really limited. Having clean commit histories makes review more efficient, so this is an area of programming I strongly encourage you to play around with - make a branch, make a few commits on it that are just to play with (simple text changes) and get used to running `git show -p HEAD~2` or similar, running `git rebase -i HEAD~4` (or similar) and playing with moving the commits around, squashing them, dropping, rewording etc, and force-pushing to your branch to see what happens. Then you'll be a :mage: - definitely worth it

**Comment by mikehardy** [comment]
Now this one took a conflict - but it's actually because we have merged another of your PRs (congratulations :-) ) - this is now the *perfect* opportunity to play with git rebase! Take the time, it's worth it

First, you will want to squash this all down to one commit:
`git rebase -i HEAD~4` - and then you will have an editor with the 4 commits, leave the first alone and change the last 3 to "squash" instead of "pick". Then edit the commit message and now you will have one commit with the new message.

Then do a `git fetch upstream` in order to get your local system up to date with upstream, and now do `git rebase upstream main`. You will have a conflict in lint.xml. Resolve it by re-applying your change, do a `git add -A` to tell git rebase you've fixed things, then `git rebase --continue`

Then `git push origin +enable-RedundantNamespace-lint` to push the changes back out to your fork on github and poof, you've done it.

At any point in the rebase you can say `git rebase --abort` to rewind to just your new squashed commit and try again fresh.

**Comment by Prince-kushwaha** [comment]
@mikehardy i made so many commit in this pull request and i do wrong squash commit can  i made another separate  pull request for this issue   and close this pull request

**Comment by Prince-kushwaha** [comment]
@mikehardy thank you for guide me

**Comment by Prince-kushwaha** [comment]
@mikehardy can i create another PR for this and close this pull request ?

**Comment by mikehardy** [comment]
No, power through it. Learn git. Learn how to force push your branch on GitHub to clean it. It's an important skill for contribution to this project

**Pull request closed without merging**